test/test_contra_env_setup.py

```diff
@@ -1,115 +1,138 @@
+_VIRSH_BIN = get_command('virsh', package='libvirt-client')
+_ANSIBLE_BIN = get_command('ansible', package='ansible')
+_QEMU_BIN = get_command('qemu-img', package='qemu-img')
+
+
+class TestMinishift(Test):
     """
-    :avocado: enable
+    class to test the contra-env-setup
     """
+
     def setUp(self):
-        # getting parameters from test_contra_env_setup.yaml
-        self.distro = self.params.get('distro', default='Fedora')
-        self.pkg_mgm = self.params.get('pkg_mgm', default='dnf')
+        """
+        configure the virtual machine that will be used in the tests
+        """
+        self.cmd_run = self.params.get('command', default=None)
         self.debug = self.params.get('debug', default=False)
         self.iso_src = self.params.get('minishift_iso_src_path',
                                        default='~/.contra-env-setup/minishift')
         self.iso_dest = self.params.get('minishift_iso_dest_path',
                                         default='~/.contra-env-setup/minishift')
-        self.cmd_run = self.params.get('command', default=None)
+        self.username = self.params.get('username', default='avocado')
+        self.password = self.params.get('password', default='avocado')
+        self.vm_name = self.params.get('vm_name', default='contra-env-setup-01')
         self.extra_cmd_run = self.params.get('extra_command', default=None)
+        self.distro = self.params.get('distro', default='Fedora')
+        self.pkg_mgm = self.params.get('pkg_mgm', default='dnf')
 
         project = 'https://github.com/CentOS-PaaS-SIG/contra-env-setup.git'
         repo = self.params.get('repo', default=project)
         branch = self.params.get('branch', default='master')
 
-        # cloning the repo defined in the parameter file
         git.get_repo(repo,
                      branch=branch,
                      destination_dir=self.workdir)
 
-        # getting the VM image qcow2
-        image = vmimage.get(self.distro, cache_dir='~/avocado/data/cache/')
+        self.image = vmimage.get(self.distro, cache_dir='~/avocado/data/cache/')
 
-        # resize storage of VM
-        qemu_img_bin = utils_path.find_command('qemu-img')
-        process.run('%s resize %s +20G' % (qemu_img_bin, image.path))
+        process.run('{qemu} resize '
+                    '{image_path} 40G'.format(qemu=_QEMU_BIN,
+                                              image_path=self.image.path))
 
-        # adding the VM image
-        self.vm.add_image(image.path, cloudinit=True, snapshot=False)
+        cmd = [_VIRTINST_BIN,
+               '--import', '--noautoconsole',
+               '--connect', 'qemu:///system',
+               '--name', self.vm_name,
+               '--disk', '%s,size=40' % self.image.path,
+               '--graphics', 'vnc',
+               '--cpu', 'host',
+               '--memory', '7168',
+               '--vcpus', '4',
+               '--os-variant', 'fedora27',
```

Comment:
shouldn't this `--os-variant` be changed according to the respective `self.distro`? no sure it would make any difference here...